### PR TITLE
docs: ✏️ Add JSDoc comment to CreateCheckpointScheduleParams

### PR DESCRIPTION
**Pull request by polymath-eric**
Adds documents explaining the meaning of null values when creating a
CheckpointSchedule

### CODE REVIEW

File: src/api/procedures/createCheckpointSchedule.ts [hunk 1]
```diff
@@ -18,8 +18,17 @@ import {
...
 
 export interface CreateCheckpointScheduleParams {
+  /**
+   * The date for when to begin creating snapshots. A null value indicates immediately
+   */
   start: Date | null;
+  /**
+   * The cadence in which to make Checkpoints. A null value indicates to create only one Checkpoint, regardless of repetitions specified
+   */
   period: CalendarPeriod | null;
```
Comment (polymath-eric) [inline src/api/procedures/createCheckpointSchedule.ts:28]: To me a null period doesn't make sense when there is an API to create a single Checkpoint.

Comment (monitz87) [reply src/api/procedures/createCheckpointSchedule.ts:28]: This allows you to schedule its creation in the future

Comment (monitz87) [reply src/api/procedures/createCheckpointSchedule.ts:28]: But yeah, null date + null period is the same as just creating a Checkpoint directly

File: src/api/procedures/createCheckpointSchedule.ts [hunk 2]
```diff
@@ -18,8 +18,17 @@ import {
 import { filterEventRecords } from '~/utils/internal';
 
 export interface CreateCheckpointScheduleParams {
+  /**
+   * The date for when to begin creating snapshots. A null value indicates immediately
```
Comment (monitz87) [inline src/api/procedures/createCheckpointSchedule.ts:22]:
```suggestion
   * The date from which to begin creating snapshots. A null value indicates immediately
```

File: src/api/procedures/createCheckpointSchedule.ts [hunk 3]
```diff
@@ -18,8 +18,17 @@ import {
...
+   */
   start: Date | null;
+  /**
+   * The cadence in which to make Checkpoints. A null value indicates to create only one Checkpoint, regardless of repetitions specified
```
Comment (monitz87) [inline src/api/procedures/createCheckpointSchedule.ts:26]: I think "cadence with which to make..." is the correct usage, but I'm actually not sure and I couldn't find it anywhere, so I leave it up to you

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/createCheckpointSchedule.ts (hunk 1 above)
Comment (polymath-eric) [reply src/api/procedures/createCheckpointSchedule.ts:28]: Oh, I didn't think about that. I guess it does give new functionality then.

File: src/api/procedures/createCheckpointSchedule.ts (hunk 3 above)
Comment (polymath-eric) [reply src/api/procedures/createCheckpointSchedule.ts:26]: I think you are right.

File: src/api/procedures/createCheckpointSchedule.ts (hunk 1 above)
Comment (polymath-eric) [reply src/api/procedures/createCheckpointSchedule.ts:29]: I moved it to a @note, + specified that use case.

**Review by monitz87 — APPROVED** [review]

**Pull request merged**